src/wp-includes/rest-api/endpoints/class-wp-rest-global-styles-controller.php

```diff
@@ -263,6 +263,23 @@ protected function prepare_item_for_database( $request ) {
 			} elseif ( isset( $existing_config['styles'] ) ) {
 				$config['styles'] = $existing_config['styles'];
 			}
+
+			// If the incoming request is going to create a new variation
+			// that is not yet registered, we register it here.
+			// This is because the variations are registered on init,
+			// but we want this endpoint to return the new variation immediately:
+			// if we don't register it, it'll be stripped out of the response
+			// just in this request (subsequent ones will be ok).
+			// Take the variations defined in styles.blocks.variations from the incoming request
+			// that are not part of the $exsting_config.
+			if ( isset( $request['styles']['blocks']['variations'] ) ) {
```

Comment:
I've noticed there's an alternative to this approach: using the `rest_insert_wp_global_styles` filter to register _any_ variation provided in the `$request`.

We wouldn't know which ones are _new_ because we'd have access to the _updated_ post not to the old one because `rest_insert_wp_global_styles` is dispatched _after_ the post was created. That could be an advantage: the current approach may end up registering a variation that is not stored in the DB if there is an error saving the post — I don't think this would be an issue, though, because the response of `update_item` won't return the incoming variation that wasn't saved to the DB.

I went back and forth on this, but I don't have a strong opinion either way. All in all, I thought collocating the code here instead of having it "disconnected" via filter was more valuable.

Reply:
I can see pros and cons to either approach and as such don't have a strong opinion either. 

That said, I think this approach is a little easier to discover so happy to go ahead with it 👍

Reply:
ok, let's go with this then. 

> Nit: The format of the multiline comment above will need updating to meet core's standards

Fixed at https://github.com/WordPress/wordpress-develop/pull/6756/commits/10a638a57c91e0ed1ad3c965c44eced4acbf8b40